scripts/deleteAndLoadSnapsets.js

```diff
@@ -0,0 +1,398 @@
+ */
+dbLoader.createFetchGpiiKeysStep = function (options, previousStep) {
+    var togo = fluid.promise();
+    var response = dbLoader.createResponseHandler(
+        dbLoader.processGpiiKeys,
+        togo,
+        "Error finding snapset Prefs Safes associated GPII Keys: "
+    );
+    var request = dbLoader.queryDatabase(
+        options.gpiiKeysViewUrl,
+        response,
+        "Error requesting GPII Keys: "
+    );
+    previousStep.then(function () { request.end(); }, dbLoader.bail);
+    return togo;
+};
+
+/*
+ * Create the step that deletes, in batch, the current snapset Prefs Safes and
+ * their associated GPII keys.
+ * @param {Promise} previousStep - Promise from a previous step whose fulfillment
+ *                                 triggers the bulk delete request.
+ * @return {Promise} - The promise asoociated with this step.
+ */
+dbLoader.createBatchDeleteStep = function (previousStep) {
+    var togo = fluid.promise();
+    var response = dbLoader.createResponseHandler(
+        function () {
+            fluid.log("Snapset Prefs Safes and associated GPII Keys deleted.");
+        },
+        togo
+    );
+    previousStep.then(
+        function () { dbLoader.doBatchDelete(response); },
+        dbLoader.bail
+    );
+    return togo;
+};
+
+/*
+ * Create the step that uploads, in batch, the new snapset Prefs Safes, their
+ * associated GPII keys, and the demo user Prefs Safes/GPII Keys.
+ * @param {Object} options - Object that has the paths to the directories that
+ *                           contain the new snapsets and demo user preferences.
+ * @param {Promise} previousStep - Promise from a previous step whose fulfillment
+ *                                 triggers this bulk upload request.
+ * @return {Promise} - The promise asoociated with this step.
+ */
+dbLoader.createBatchUploadStep = function (options, previousStep) {
+    var togo = fluid.promise();
+    var buildData = dbLoader.getDataFromDirectory(options.buildDataDir);
+    var demoUserData = dbLoader.getDataFromDirectory(options.demoUserDir);
+    var allData = buildData.concat(demoUserData);
+    var response = dbLoader.createResponseHandler(
+        function () {
+            fluid.log ("Bulk loading of build data from '" + options.buildDataDir + "'");
+        },
+        togo
+    );
+    var request = dbLoader.createBulkDocsRequest(allData, response, options);
+    previousStep.then(request, dbLoader.bail);
+    return togo;
+};
+
+/*
+ * Create the steps to load the static data, find and delete the current snapset
+ * Prefs Safes and the GPII keys, and then load the latest snapset Prefs Safes
+ * and their Keys, and the demo user Prefs Safes and their GPII Keys.  After all
+ * steps are configured and connected, trigger the first one.
+ */
+dbLoader.orchestrate = function () {
+    var options = dbLoader.getOptions();
+    var lastStep;
+    var firstStep = dbLoader.createFetchSnapsetsStep(options);
```

Comment:
Use fluid.promise.sequence to orchestrate rather than doing this all by hand inside the functions using the "previousStep" pattern